lib/beaker/hypervisor/lxc.rb

```diff
@@ -0,0 +1,102 @@
+module Beaker
+  class Lxc < Beaker::Hypervisor
+
+  def initialize(hosts, options)
+    require 'lxc'
+
+    @options = options
+    @logger  = options[:logger]
+    @hosts   = hosts
+  end
+
+  def root_password
+    'root'
+  end
+
+  def provision
+   @hosts.each do | host |
+
+      @logger.notify "Provisioning lxc"
+      container = LXC::Container.new(host)
+
+      template = host['template']
+      arch = host['arch'] || "amd64"
+
+      if match = template.match(/(.*):(.*)/)
+        image = match.captures[0]
+        release = match.captures[1]
+      end
+
+      @logger.notify "Creating lxc #{host} with #{template}"
+      container.create("download", nil, {}, 0, ["-d", "#{image}", "-r", "#{release}", "-a", "#{arch}"])
+
+      @logger.notify "Setting up config #{host}"
+      container.set_config_item('lxc.cap.drop', '')
+      container.save_config
+
+      @logger.notify "Starting Lxc #{host}"
+      container.start
+
+      @logger.notify "Attaching Lxc to #{host}"
+        container.attach(:wait => true) do
+          case host['platform']
+          when /ubuntu/, /debian/
+            puts `apt-get update`
```

Comment:
@anodelman nope, this will run inside container, #41 line show that its getting attach to the container.